internal/handlers/initrd.go

```diff
@@ -71,11 +81,49 @@ func (h *initrdHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	fileName := fmt.Sprintf("%s-initrd.img", imageID)
-	w.Header().Set("Content-Disposition", fmt.Sprintf("attachment; filename=%s", fileName))
+	aReader := initrdReader
+
+	// if not s390x and addrsize=1 than regular processing
+	if addrsize != "" && addrsize == "1" {
+		// calculate the addrsize and return initrd.addrsize file
+		// get current addrsize file from iso
+		fileName = fmt.Sprintf("%s-initrd.addrsize", imageID)
+		addrsizeReader, err := isoeditor.GetFileFromISO(isoPath, initrdAddrsizePathInISO)
+		if err != nil {
+			log.Errorf("Error retrieving initrd.addsize file: %v, isoPath; %s\n", err, isoPath)
+			httpErrorf(w, http.StatusInternalServerError, "Failed to get initrd: %v", err)
+			return
+		}
+
+		// get the size of the initrd including the embedded ignition
+		sizeOfInitrd, err := initrdReader.Seek(0, io.SeekEnd)
+		if err != nil {
+			httpErrorf(w, http.StatusInternalServerError, "Failed to determine size of initrd: %v", err)
+			return
+		}
+		buf := new(bytes.Buffer)
+		binary.Write(buf, binary.BigEndian, sizeOfInitrd)
+
+		// create overlay reader for initrdsize
+		overlay_addrsize := overlay.Overlay{
+			Reader: bytes.NewReader(buf.Bytes()),
+			Offset: 8,
+			Length: 8,
+		}
+		aReader, err = overlay.NewOverlayReader(addrsizeReader, overlay_addrsize)
+		if err != nil {
+			httpErrorf(w, http.StatusInternalServerError, "Failed to create overlay reader for initrd.addrsize: %v", err)
+			return
+		}
+
+		defer addrsizeReader.Close()
```

Comment:
Please move all the code my comment is replying to to its own separate, well-named function